src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ReferralTest.kt

```diff
@@ -41,4 +44,30 @@ internal class ReferralTest {
       assertThat(referral.currentAssignee).isEqualTo(currentAssignee)
     }
   }
+
+  @Nested inner class ActionPlans {
+    @Test
+    fun `approvedActionPlan returns null if there are no approved action plan`() {
+      val referral = referralFactory.createSent(
+        actionPlans = mutableListOf(
+          actionPlanFactory.createSubmitted()
+        )
+      )
+
+      assertThat(referral.approvedActionPlan).isNull()
+    }
+
+    @Test
+    fun `approvedActionPlan returns most recent approved action plan`() {
```

Comment:
Are there a couple of test scenarios here, one for latest approved and then one for latest created when non approved

Reply:
the non-appoved action plans are covered by the other getter 'currentActionPlan', which does not filter on approval.